src/core/proj/qgscoordinatetransform.cpp

```diff
@@ -896,7 +899,16 @@ bool QgsCoordinateTransform::setFromCache( const QgsCoordinateReferenceSystem &s
   const QList< QgsCoordinateTransform > values = sTransforms.values( qMakePair( sourceKey, destKey ) );
   for ( auto valIt = values.constBegin(); valIt != values.constEnd(); ++valIt )
   {
-    if ( ( *valIt ).coordinateOperation() == coordinateOperationProj && ( *valIt ).allowFallbackTransforms() == allowFallback )
+    if ( ( *valIt ).coordinateOperation() == coordinateOperationProj
+         && ( *valIt ).allowFallbackTransforms() == allowFallback
+
+         // careful here, nan != nan, so we need to explicitly handle the case when both crses have nan coordinateEpoch
```

Comment:
for readability, couldn't we have a qgsNanCompatibleEquals(x, y) inline function that would do that ?

Reply:
Good point, will do that!

Reply:
Ok, done!